### PR TITLE
interfaces: add common STD_CARGO_KG to mass

### PR DESCRIPTION
**Pull request by jyoung8607**
**Description**

Add STD_CARGO_KG to vehicle mass once, instead of every single time we set vehicle mass.

* ~~special handling for Civic in `selfdrive/car/__init__.py` to prevent circular references~~
* special handling for comma body in `selfdrive/car/interfaces.py` so it doesn't carry a human around

GitHub will tell you this PR is +5, but if you listen to your heart 👩‍🎤 it's -140.

**Verification**

```python
#!/usr/bin/env python3

from selfdrive.car import gen_empty_fingerprint
from selfdrive.car.car_helpers import interfaces
from selfdrive.car.fingerprints import all_known_cars

if __name__ == "__main__":
  for car in all_known_cars():
    CarInterface, CarController, CarState = interfaces[car]
    car_fw = []
    fingerprints = gen_empty_fingerprint()
    CP = CarInterface.get_params(car, fingerprints, car_fw, experimental_long=False, docs=False)
    print(f"{CP.carFingerprint}: {CP.mass=}")
```

[mass_before.txt](https://github.com/commaai/openpilot/files/12353665/mass_before.txt)
[mass_after.txt](https://github.com/commaai/openpilot/files/12353666/mass_after.txt)

**Differences**

Some cars with mass listed in pounds (and therefore multiplied by `CV.LB_TO_KG`) change very slightly because the cereal Float32 conversion is now done before adding `STD_CARGO_KG` instead of afterward.

```
GMC ACADIA DENALI 2018: CP.mass=2110.486083984375
GMC ACADIA DENALI 2018: CP.mass=2110.48583984375
LEXUS RX 2016: CP.mass=2168.545654296875
LEXUS RX 2016: CP.mass=2168.5458984375
LEXUS RX 2020: CP.mass=2168.545654296875
LEXUS RX 2020: CP.mass=2168.5458984375
LEXUS RX HYBRID 2017: CP.mass=2168.545654296875
LEXUS RX HYBRID 2017: CP.mass=2168.5458984375
LEXUS RX HYBRID 2020: CP.mass=2168.545654296875
LEXUS RX HYBRID 2020: CP.mass=2168.5458984375
TOYOTA ALPHARD 2020: CP.mass=2088.713623046875
TOYOTA ALPHARD 2020: CP.mass=2088.71337890625
TOYOTA ALPHARD HYBRID 2021: CP.mass=2088.713623046875
TOYOTA ALPHARD HYBRID 2021: CP.mass=2088.71337890625
```

Several HKG vehicles weren't properly including STD_CARGO_KG prior to this PR. In this case, we're essentially fixing a bug.
```
GENESIS G90 2017: CP.mass=2200.0
GENESIS G90 2017: CP.mass=2336.0
HYUNDAI SONATA 2019: CP.mass=2039.80322265625
HYUNDAI SONATA 2019: CP.mass=2175.80322265625
HYUNDAI TUCSON 2019: CP.mass=1596.643798828125
HYUNDAI TUCSON 2019: CP.mass=1732.643798828125
HYUNDAI VELOSTER 2019: CP.mass=1613.88037109375
HYUNDAI VELOSTER 2019: CP.mass=1749.88037109375
KIA FORTE E 2018 & GT 2021: CP.mass=1613.88037109375
KIA FORTE E 2018 & GT 2021: CP.mass=1749.88037109375
KIA K5 2021: CP.mass=1464.1949462890625
KIA K5 2021: CP.mass=1600.1949462890625
KIA K5 HYBRID 2020: CP.mass=1464.1949462890625
KIA K5 HYBRID 2020: CP.mass=1600.1949462890625
KIA OPTIMA 4TH GEN FACELIFT: CP.mass=1613.88037109375
KIA OPTIMA 4TH GEN FACELIFT: CP.mass=1749.88037109375
KIA OPTIMA 4TH GEN: CP.mass=1613.88037109375
KIA OPTIMA 4TH GEN: CP.mass=1749.88037109375
KIA OPTIMA HYBRID 2017 & SPORTS 2019: CP.mass=1613.88037109375
KIA OPTIMA HYBRID 2017 & SPORTS 2019: CP.mass=1749.88037109375
```

### CODE REVIEW

**Comment by jyoung8607** [comment]
Additional fix: the base mass for `HYUNDAI SONATA 2019` was wrong. The value of 2039 kg appears to be a gross (maximum) weight, it should be more like 1536 kg (average of curb weights according to the Internets). This is a pretty hefty difference and it might have had a negative effect on params learning/tuning. This had to be addressed in this PR; it's one of the cars that wasn't adding STD_CARGO_KG in the past and adding it now would exacerbate the problem.

File: selfdrive/car/__init__.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def gen_empty_fingerprint():
 # FIXME: hardcoding honda civic 2016 touring params so they can be used to
 # scale unknown params for other cars
 class CivicParams:
-  MASS = 1326. + STD_CARGO_KG
+  MASS = 1326.
+  MASS_WITH_CARGO = MASS + STD_CARGO_KG
```
Comment (adeebshihadeh) [inline selfdrive/car/__init__.py:47]: is MASS only used to set the civic weight in the honda interface? if so, think it's fine to just have that constant twice and delete this MASS

**Comment by sshane** [comment]
You can fix any incorrect masses in a new PR! Like to keep refactors free from behavior-changes

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (jyoung8607) [reply selfdrive/car/__init__.py]: The Civic car port [uses three constants](https://github.com/commaai/openpilot/blob/4c1b8b71d096e7658fe56598944e3f8bd3e347f0/selfdrive/car/honda/interface.py#L91-L115) from this class. IMO it doesn't make sense to duplicate the mass and not the other parameters, and having the Civic port use this special snowflake construct seems to have more costs than benefits. LMK how you like the most current version.

**Comment by jyoung8607** [comment]
> You can fix any incorrect masses in a new PR! I like to keep refactors free from behavior-changes, no matter small

Agreed, I didn't realize there would be functional changes until I fully validated the outcome.

The mass fixes are set up in #29418, once that's merged I'll rebase and revalidate this PR.

**Comment by jyoung8607** [comment]
This version of the refactor pushes the Honda Civic parameters down into the car port, so it's no longer a special case with the potential to create circular references. I'm happier with this, @adeebshihadeh hopefully this resolves your concern.

I revalidated all cars after #29418 and found the only remaining differences trace back to floating-point precision issues, which have been previously identified and can be safely ignored.

[mass2_before.txt](https://github.com/commaai/openpilot/files/12361277/mass2_before.txt)
[mass2_after.txt](https://github.com/commaai/openpilot/files/12361278/mass2_after.txt)

```
GMC ACADIA DENALI 2018: CP.mass=2110.486083984375
GMC ACADIA DENALI 2018: CP.mass=2110.48583984375
LEXUS RX 2016: CP.mass=2168.545654296875
LEXUS RX 2016: CP.mass=2168.5458984375
LEXUS RX 2020: CP.mass=2168.545654296875
LEXUS RX 2020: CP.mass=2168.5458984375
LEXUS RX HYBRID 2017: CP.mass=2168.545654296875
LEXUS RX HYBRID 2017: CP.mass=2168.5458984375
LEXUS RX HYBRID 2020: CP.mass=2168.5458984375
LEXUS RX HYBRID 2020: CP.mass=2168.545654296875
TOYOTA ALPHARD 2020: CP.mass=2088.713623046875
TOYOTA ALPHARD 2020: CP.mass=2088.71337890625
TOYOTA ALPHARD HYBRID 2021: CP.mass=2088.713623046875
TOYOTA ALPHARD HYBRID 2021: CP.mass=2088.71337890625
```

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (jyoung8607) [reply selfdrive/car/__init__.py]: It also wasn't obvious that changing Honda Civic mass or wheelbase parameters would indirectly alter params for *every other car* known to openpilot.

Perhaps we accept that they're decoupled, and rename the CivicParams class to BenchmarkParams and treat it as a snapshot-in-time of a nonspecific idealized/typical car?

Comment (sshane) [reply selfdrive/car/__init__.py]: We can do that in another PR, will make the change easier to think about

**Review by adeebshihadeh — APPROVED** [review]
LGTM, nice cleanup! @sshane anything blocking this?

**Comment by jyoung8607** [comment]
> LGTM, nice cleanup! @sshane anything blocking this?

I believe we were in agreement on that last item of feedback, I just haven't done it yet.

**Comment by sshane** [comment]
https://github.com/commaai/openpilot/pull/29459

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/__init__.py]: https://github.com/commaai/openpilot/pull/29459

**Pull request merged**